library/src/main/java/com/novoda/downloadmanager/DownloadMigrationService.java

```diff
@@ -0,0 +1,27 @@
+package com.novoda.downloadmanager;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.WorkerThread;
+
+public interface DownloadMigrationService {
+
+    MigrationFuture startMigration(NotificationChannelCreator notificationChannelCreator, NotificationCreator<MigrationStatus> notificationCreator);
```

Comment:
The builder allowed to optionally inject custom implementations, now you are forcing the user to pass both when he might actually just want to use a custom notificationChannelCreator, not both. I'm not sure this change is for better.